android/src/main/java/nl/kega/reactnativeemdk/BarcodeScannerThread.java

```diff
@@ -115,280 +117,269 @@ private void destroyScanner() {
-                            break;
-                            case "ean13":  
-                                this.scanner_config.decoderParams.ean13.enabled = true;
-                            break;
-                            case "codabar":  
-                                this.scanner_config.decoderParams.codaBar.enabled = true;
-                            break;
-                            case "code11":  
-                                this.scanner_config.decoderParams.code11.enabled = true;
-                            break;
-                            case "code39":  
-                                this.scanner_config.decoderParams.code39.enabled = true;
-                            break;
-                            case "code93":  
-                                this.scanner_config.decoderParams.code93.enabled = true;
-                            break;
-                            case "code128":  
-                                this.scanner_config.decoderParams.code128.enabled = true;
-                            break;
-                            case "qrcode":  
-                                this.scanner_config.decoderParams.qrCode.enabled = true;
-                            break;
-                            case "dutchpostal":  
-                                this.scanner_config.decoderParams.dutchPostal.enabled = true;
-                            break;
-                            
-                        }
-                        
-                    }
-                }else{
-                    this.scanner_config.decoderParams.ean8.enabled = true;
-                    this.scanner_config.decoderParams.ean13.enabled = true;
-                    this.scanner_config.decoderParams.codaBar.enabled = true;
-                    this.scanner_config.decoderParams.code11.enabled = true;
-                    this.scanner_config.decoderParams.code39.enabled = true;
-                    this.scanner_config.decoderParams.code93.enabled = true;
-                    this.scanner_config.decoderParams.code128.enabled = true;
-                    this.scanner_config.decoderParams.qrCode.enabled = true;
-                    this.scanner_config.decoderParams.dutchPostal.enabled = true;
-                }
-
-                this.scanner_config.scanParams.decodeHapticFeedback = true;
-
-                this.scanner.setConfig(this.scanner_config);
-
+                this.scanner.setConfig(configureScanner(scanner));
                 this.scanner.read();
-
             }
-
         } catch (ScannerException e) {
-            Log.e("[BarcodeScanner]", "Read error: " + e);
+            Log.e(TAG, "Read error: " + e);
         } catch (NullPointerException e) {
-        
+            Log.e(TAG, "Read null pointer error: " + e);
         }
-
     }
 
+    /**
+     * Sets TriggerType to SOFT_ALWAYS and calls Scanner.read().
+     * This method should be used when only using a software button to

```

Comment:
Maybe "This method should be used when a software button is the only trigger for the scanner."